crates/ruff_cli/src/diagnostics.rs

```diff
@@ -428,20 +428,80 @@ pub(crate) fn lint_stdin(
                 flags::FixMode::Diff => {
                     // But only write a diff if it's non-empty.
                     if !fixed.is_empty() {
-                        let text_diff = TextDiff::from_lines(
-                            source_kind.source_code(),
-                            transformed.source_code(),
-                        );
-                        let mut unified_diff = text_diff.unified_diff();
-                        if let Some(path) = path {
-                            unified_diff
-                                .header(&fs::relativize_path(path), &fs::relativize_path(path));
-                        }
+                        match transformed.as_ref() {
+                            SourceKind::Python(_) => {
```

Comment:
nit: This is repeated for stdout as well but I think the only difference is that the path is optional. Could we combine them under `SourceKind` in the same way as done in `FixMode::Apply`? That would avoid code duplication.